### ISSUE TITLE
`TestRunner` could be improved to report more results, better

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#561]
Some code that would go in #560 but I decided it doesn't fit there.

```dart
// TODO: Use this
class SingleTargetRunResult {
  SingleTargetRunResult({
    required this.target,
    required this.device,
    required this.plannedRuns,
    this.buildPassed = false,
    required this.runs,
  });

  final String target;
  final Device device;
  bool buildPassed;

  final int plannedRuns;
  final List<RunStatus> runs;
}
```

And then `TestRunner.run()`'s signature would be:

```dart
Future<List<SingleTargetRunResult>> run() {}
```

Benefits:
- This would make code in `DriveCommand` cleaner (it'd be moved to `TestRunner`)
- All the complexity would be contained inside `TestRunner` and properly tested
- We could print a nice test result summary, similar to what `spec` does:

  ![Screenshot 2022-11-03 at 6 32 54 PM](https://user-images.githubusercontent.com/40357511/199793490-b73fbe81-faf6-48a9-9e5c-98ed5c921d32.png)

  Without colors it looks like:
  
  ```
   PASS  test/command_runner_test.dart
   PASS  test/features/drive/test_runner_test.dart
   PASS  test/features/drive/drive_command_test.dart
   PASS  test/features/drive/device_finder_test.dart
   PASS  test/features/drive/test_finder_test.dart
   PASS  test/common/artifacts_repository_test.dart
  ```

  I'd like to print something like:

    ```
   PASS  integration_test/core_cutter_test.dart.dart (build passed, 10/10 runs passed)
   FAIL  integration_test/dual_cbr_test.dart  (build failed)
   FAIL  integration_test/hsv_test.dart (build passed, 9/10 runs passed)
   CANC  integration_test/srd_test.dart  (build passed, 3/10 runs passed)
  ```

**Comment by jBorkowska** [comment]
Is this a place to add the error messages for why a test failed? Or would it be kept separate?

**Comment by bartekpacia** [comment]
No. To find you why a test failed, you'd have to still look in the logs.

**Comment by jBorkowska** [comment]
From my perspective: I need a place, where I can easily find info about which test failed and why. Logs are quite hard to read (at least now), that's why I'm asking.

**Comment by jakubfijalkowski** [comment]
Jest prints, apart from the passed/failed marker, as summary of failed test result at the end of the output. This is, IMO, quite good middle-ground between being fancy (so a pass/fail colored marker) and being useful. If everything passes you have clean output, if something fails, you see what failed right there, without looking for it.

**Comment by bartekpacia** [comment]
Yeah, we're aware of it. We hope to improve this!

For now, you can use `flutter logs` while running `patrol test` to view Dart-side logs.

**Comment by bartekpacia** [comment]
This is a possible duplicate of #870.
